migrator/migrations/m_197_to_m_198_add_oidc_claim_mappings/store/authproviders/store.go

```diff
@@ -0,0 +1,256 @@
+package authproviders
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/jackc/pgx/v5"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	schemaPkg "github.com/stackrox/rox/migrator/migrations/m_197_to_m_198_add_oidc_claim_mappings/schema"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/postgres"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	"github.com/stackrox/rox/pkg/search"
+	pgSearch "github.com/stackrox/rox/pkg/search/postgres"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+const (
+	// using copyFrom, we may not even want to batch.  It would probably be simpler
+	// to deal with failures if we just sent it all.  Something to think about as we
+	// proceed and move into more e2e and larger performance testing
+	batchSize       = 10000
+	batchAfter      = 100
+	deleteBatchSize = 5000
+)
+
+var (
+	log    = logging.LoggerForModule()
+	schema = schemaPkg.AuthProvidersSchema
+)
+
+type storeType = storage.AuthProvider
+
+// Store is the interface to interact with the storage for storage.AuthProvider
+type Store interface {
+	UpsertMany(ctx context.Context, objs []*storeType) error
+	Walk(ctx context.Context, fn func(obj *storeType) error) error
+}
+
+type storeImpl struct {
+	mutex sync.RWMutex
+	db    postgres.DB
+}
+
+// New returns a new Store instance using the provided sql instance.
+func New(db postgres.DB) Store {
+	return &storeImpl{
```

Comment:
I don't really have an idea: do we use the generic store now in migrations or not? It seems like some migrations do, while others do not? I don't have a strong feeling either way, but this is probably something worthwhile figuring out and documenting.

Maybe @dashrews78 has an opinion / guidance on this.

Reply:
@dhaus67 @ivan-degtiarenko We can and should use the generic store here with a couple of caveats.

1.  Use an interface in the migration to expose only a subset of the generic store to the migration.  The reason for this is we don't want to use the search features as those can change over time.  But the walk should be safe to use.  Similarly the Upserts would be safe because when using the generic store we have to implement the sub functions the upsert uses anyway.
2. Remove the references to SAC.

This is somewhat mentioned in the `README`
```
   Now that stores are built upon a generic store you would need to make a copy of the store and remove
   references to SAC and such.  We should be careful not to rely on the search framework for queries as those
   could change over time.  Additionally the stores should use the frozen schemas just as any other migration would
   to ensure isolation. 
```

#7893 is a stale hackathon PR, but I find it to be a clean example of what a using a generic store would look like.  migration 196 also uses the generic store, but it exposes all the power of the generic store to the migration.